### DISCUSSION TITLE
Mapbox building layer overlaid ontop of deck mvt layers

### DISCUSSION THREAD

**Discussion by dkspec** [visgl/deck.gl#6241] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/61771607/134868377-a2d4a8d0-d410-4afa-b92f-930be30634e4.png)

I have MVT Layers coming in as a deck layer which gets rendered as part of deck instance.
Mapbox layer is being rendered with Static Map and gets overlaid ontop of this deck instance.

When I am at zoomlevel 16 and greater, building layers from mapbox gets stays on top of deck layers.
Is there a way to force all the mapbox layers to be drawn underneath the deck layers? Without interleaving deck layer into mapbox gl js.

Thanks

      <DeckGL
        style={mapStyles}
        ref={deckRef}
        controller={true}
        onWebGLInitialized={(context) => setGLContext(context)}
        glOptions={{
          /* To render vector tile polygons correctly */
          stencil: true,
        }}
        initialViewState={deckProps.initialViewState}
        layers={[new MVTLayer()]} // where mvt layer is brought in
      >
        {glContext && (
          <Fragment>
            <StaticMap
              ref={mapRef}
              gl={glContext}
              mapStyle={mapbox://styles/mapbox/light-v9}
              mapboxApiAccessToken={process.env.REACT_APP_MAPBOX_ACCESS_TOKEN}
              onLoad={onLoad}
              mapOptions={deckProps.mapOptions || {}}
            >
              {mapboxSourceLayers}
              {mapboxLayers}
            </StaticMap>
          </Fragment>
        )}
      </DeckGL>

**Comment by chrisgervang** [comment, marked as answer]
> force all the mapbox layers to be drawn underneath the deck layers? Without interleaving deck layer into mapbox gl js.

One way to have mapbox underneath is to not share the glContext. [example](https://github.com/visgl/deck.gl/blob/8.5-release/examples/get-started/react/mapbox/app.js). Are you looking for an [overlaid](https://deck.gl/docs/get-started/using-with-map) implementation? 

You would only need interleaving if you want Z to mix, or if you want to capture video/screenshots of your map.
